src/utils.jl

```diff
@@ -24,17 +24,35 @@ stdscore(a::Measurement{S}, b::Measurement{T}) where {S<:AbstractFloat,T<:Abstra
     stdscore(a - b, zero(promote_type(S, T)))
 
 # Weighted Average with Inverse-Variance Weighting
+measurement_eltype(::Measurement{T}) where T = T
 """
-    weightedmean(iterable) -> measurement(weighted_mean, standard_deviation)
+    weightedmean(iterable; dims::Union{Colon,Int}=:)
 
 Return the weighted mean of measurements listed in `iterable`, using
-inverse-variance weighting.  NOTA BENE: correlation is not taken into account.
-"""
-function weightedmean(iterable)
-    v = [el.val for el in iterable]
-    w = [inv(el.err)^2 for el in iterable]
-    invsumw = inv(sum(w))
-    return measurement(dot(v, w)*invsumw, sqrt(invsumw))
+inverse-variance weighting.  `dims` is a vector specifying the
+dimensions to reduce over, or `:` to reduce over all dimensions.
+NOTA BENE: correlation is not taken into account.
+"""
+function weightedmean(iterable; dims=:)
```

Comment:
The typing of `dims` doesn't match the docstring
```suggestion
function weightedmean(iterable; dims::Union{Colon,Int}=:)
```